lms/djangoapps/completion/models.py

```diff
@@ -147,3 +274,41 @@ def __unicode__(self):
             block_key=self.block_key,
             completion=self.completion,
         )
+
+
+class AggregateCompletion(TimeStampedModel):
+    """
+    Aggregators are blocks that contain other blocks, are not themselves completable,
+    and are considered complete when all descendant blocks are complete.
+    """
+    id = BigAutoField(primary_key=True)  # pylint: disable=invalid-name
+    user = models.ForeignKey(User)
+    course_key = CourseKeyField(max_length=255)
+    aggregation_name = models.CharField(max_length=255)
+    block_key = UsageKeyField(max_length=255)
+    earned = models.FloatField(validators=[validate_positive_float])
+    possible = models.FloatField(validators=[validate_positive_float])
+    percent = models.FloatField(validators=[validate_percent])
+
+    objects = AggregateCompletionManager()
+
+    class Meta(object):
+        index_together = [
+            ('user', 'aggregation_name', 'course_key', 'block_key'),
+            ('course_key', 'aggregation_name', 'block_key', 'percent'),
+        ]
+
+        unique_together = [
+            ('course_key', 'block_key', 'user', 'aggregation_name')
+        ]
+
+    def __unicode__(self):
+        return 'AggregationCompletion: {username}, {course_key}, {block_key}: {percent}'.format(
+            username=self.user.username,
+            course_key=self.course_key,
+            block_key=self.block_key,
+            percent=self.percent,
+        )
+
+
+pre_save.connect(CompletionManager.pre_save, AggregateCompletion, dispatch_uid="completion.models.AggregateCompletion")
```

Comment:
Use `AggregateCompletionManager.pre_save` instead of `CompletionManager.pre_save`